client/src/templates/TemplatePage/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { Link } from 'react-router-dom';
+
+/* Assets */
+import PoapLogo from '../../images/POAP.svg';
+
+// components
+import { NavigationMenu } from '../../backoffice/Main';
+import { TemplatesFilters } from './components/TemplatesFilters';
+import { TemplatesTable } from './components/TemplatesTable';
+
+// helpers
+import { useAsync } from '../../react-helpers';
+
+// api
+import { getTemplates, TemplateResponse } from '../../api';
+
+type PaginateAction = {
+  selected: number;
+};
+
+export const TemplatePage = () => {
+  // state hooks
+  const [name, setName] = useState<string>('');
+  const [page, setPage] = useState<number>(0);
+  const [limit, setLimit] = useState<number>(10);
+  const [total, setTotal] = useState<number>(0);
```

Comment:
If those states will only be `string` or `number` we can avoid typings due to  type inference